src/complex/Pipeline/PipelineFilter.cpp

```diff
@@ -168,6 +248,162 @@ bool PipelineFilter::execute(DataStructure& data, const std::atomic_bool& should
   return result.result.valid();
 }
 
+std::vector<DataPath> PipelineFilter::getCreatedPaths() const
+{
+  return m_CreatedPaths;
+}
+
+namespace
+{
+/**
+ * @brief Checks if one path could be a rename of another path.
+ * Returns false if the lengths do not match or more than one segment has a different name.
+ * @param path1
+ * @param path2
+ * @return bool
+ */
+bool checkIfRename(const DataPath& path1, const DataPath& path2)
+{
+  if(path1.getLength() != path2.getLength())
+  {
+    return false;
+  }
+
+  usize count = path1.getLength();
+  usize numDifferences = 0;
+  for(size_t i = 0; i < count; i++)
+  {
+    if(path1[i] != path2[i])
+    {
+      numDifferences++;
+    }
+  }
+
+  return numDifferences == 1;
+}
+
+/**
+ * @brief Finds and returns the overlap between two collections of DataPaths.
+ * @param group1
+ * @param group2
+ * @return std::vector<DataPath>
+ */
+std::vector<DataPath> getOverlap(const std::vector<DataPath>& group1, const std::vector<DataPath>& group2)
+{
+  std::vector<DataPath> overlap;
+  std::set_intersection(group1.begin(), group1.end(), group2.begin(), group2.end(), std::inserter(overlap, overlap.begin()));
+  return overlap;
+}
+
+/**
+ * @brief Removes the overlap between two groups of DataPaths
+ * @param group1
+ * @param group2
+ */
+void removeOverlap(std::vector<DataPath>& group1, std::vector<DataPath>& group2)
+{
+  auto overlap = getOverlap(group1, group2);
+  for(const auto& value : overlap)
+  {
+    std::remove(group1.begin(), group1.end(), value);
```

Comment:
If `std::remove` is intending to actually remove items from the vector, you will also need a call to `erase`. This is also the source of the `nodiscard` warnings.